src/dropdown/script.js

```diff
@@ -0,0 +1,43 @@
+window.addEventListener("hashchange", () => {
+  handleRoute(window.location.hash);
+});
+
+const CONTAINS_PATH_TEXT = "Contains current page link";
+let shownBlock = 0;
+
+function handleRoute(newPage) {
+  // Close all open popovers, would be unneeded if we actually had navigated to a new page
+  document.querySelectorAll(":popover-open").forEach((popover) => {
+    popover.hidePopover();
+  });
+
+  // Replace the content
+  const heading = document.getElementById("id-page-title");
+  heading.innerText = newPage;
+
+  const contentBlocks = document.querySelectorAll("[data-article]");
+  shownBlock = (shownBlock + 1) % contentBlocks.length;
+
+  contentBlocks.forEach((block, index) => {
+    if(index === shownBlock) block.removeAttribute('hidden');
+    else block.setAttribute('hidden', '');
+  });
+
+  const formerLink = document.querySelector("[aria-current]");
+  if (formerLink) formerLink.removeAttribute("aria-current");
+
+  const linkToRoute = document.querySelector(`[href="${newPage}"]`);
```

Comment:
Not sure what's happening, but this line seems to be crashing in Safari. I'll look into this more.